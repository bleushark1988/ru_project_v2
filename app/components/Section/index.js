import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

const Section = styled.section`
  margin: 3em auto;

  &:first-child {
    margin-top: 0;
  }
  ${props =>
    props.textCenter &&
    css`
      text-align: center;
    `};
`;

Section.propTypes = {
  textCenter: PropTypes.bool,
};

export default Section;
