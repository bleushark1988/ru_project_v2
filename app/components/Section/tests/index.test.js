import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Section from '../index';

describe('<Section />', () => {
  it('should render an <section> tag', () => {
    const renderedComponent = shallow(<Section />);
    expect(renderedComponent.type()).toEqual('section');
  });

  it('should render text center aligned when textCenter props is enabled', () => {
    const renderedComponent = renderer.create(<Section textCenter />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
    expect(renderedComponent).toHaveStyleRule('text-align', 'center');
  });

  it('should have a className attribute', () => {
    const renderedComponent = shallow(<Section />);
    expect(renderedComponent.prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Section id={id} textCenter />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const renderedComponent = shallow(<Section attribute="test" />);
    expect(renderedComponent.prop('attribute')).toBeUndefined();
  });
});
