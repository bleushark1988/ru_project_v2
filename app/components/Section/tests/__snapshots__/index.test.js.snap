// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should render text center aligned when textCenter props is enabled 1`] = `
.c0 {
  margin: 3em auto;
  text-align: center;
}

.c0:first-child {
  margin-top: 0;
}

<section
  className="c0"
/>
`;
